views/js/runner/runnerComponent.js

```diff
@@ -0,0 +1,249 @@
+                            resolve();
+                        });
+                        runner.loadItem(itemRef);
+                    });
+                });
+            }
+        };
+
+        /**
+         * @typedef {runner} runnerComponent
+         */
+        var runnerComponent = component(runnerComponentApi, defaults)
+            .setTemplate(componentTemplate)
+            .on('init', function () {
+                var self = this;
+                _.forEach(requiredOptions, function (name) {
+                    if (!self.hasOption(name)) {
+                        throw new TypeError('Missing required option ' + name);
+                    }
+                });
+            })
+            .on('destroy', function () {
+                if (runner) {
+                    runner.destroy();
+                }
+            })
+            .after('destroy', function () {
+                this.removeAllListeners();
+                runner = null;
+            })
+            .on('render', function () {
+                var self = this;
+                var plugins = [];
+                var initPromises = [];
+
+                self.hide();
+
+                if (self.hasOption('providers')) {
+                    initPromises.push(
+                        loadModules(providerLoaderFactory, self.getOption('providers'))
+                            .then(registerProviders)
+                    );
+                }
+
+                if (self.hasOption('plugins')) {
+                    initPromises.push(
+                        loadModules(pluginLoaderFactory, self.getOption('plugins'))
+                            .then(function (loadedPlugins) {
+                                plugins = loadedPlugins;
+                            })
+                    );
+                }
+
+                Promise.all(initPromises).then(function () {
+                    var runnerConfig = _.omit(self.config, ['plugins', 'providers']);
+                    runnerConfig.renderTo = self.getElement();
+
+                    runner = runnerFactory(runnerConfig.provider, plugins, runnerConfig)
+                        .on('error', function (err) {
+                            self.trigger('error', err);
+                        })
+                        .on('ready', function () {
+                            _.defer(function () {
+                                self.show();
+                                self.setState('ready');
+                                self.trigger('ready');
+
+                                resolver.resolve(runner);
+                            });
+                        })
+                        .after('destroy', function () {
+                            this.removeAllListeners();
+                        });
+
+                    self.trigger('runner', runner);
+
+                    runner.init();
+                });
+            });
+
+        try {
```

Comment:
Instead of the artifact method `whenReady`, just wrap the `init` in a `_.deferr` so implementors can listen for the `init` event. 
If using a component, I'd prefer the same patterns are used (just for consitency matters) or I am missing something ?